App/Redux/TextileRedux.js

```diff
@@ -76,39 +76,38 @@ export const toggleVerboseUi = state =>
 export const handleImageAdded = (state, {uri, thread, hash, remotePayloadPath}) => {
   const existing = state.camera && state.camera.processed ? state.camera.processed : []
   const processed = [...existing, uri]
-  const items = [{ thread, hash, remotePayloadPath, state: 'pending' }, ...state.images.items]
+  const items = [{ thread, hash, remotePayloadPath, state: 'pending', remainingUploadAttempts: 3 }, ...state.images.items]
   return state.merge({ images: { items }, camera: {processed} })
 }
 
-export const handleImageProgress = (state, {data}) => {
-  const { id, progress } = data
-  // The upload library we're using returns float 0.0 - 100.0
-  const fractionalProgress = progress / 100.0
+export const handleImageUploadRetried = (state, {hash}) => {
   const items = state.images.items.map(item => {
-    if (item.hash === id) {
-      return {...item, state: 'processing', progress: fractionalProgress}
+    if (item.hash === hash) {
+      return {...item, state: 'pending'}
```

Comment:
This diff ended up really confusing, but anyway, change the image state back to 'pending' when we start to retry the upload. This way we'll see the spinning pie and pie progress during retries.